src/chunk.c

```diff
@@ -235,16 +246,17 @@ calculate_and_set_new_chunk_interval(Hypertable *ht, Point *p)
  * slices/constraints that overlap with our tentative hypercube, i.e., they
  * fully collide. We resolve those collisions by cutting the hypercube.
  */
-static bool
+static ChunkResult
 do_collision_resolution(ChunkScanCtx *scanctx, Chunk *chunk)
 {
 	Hypercube  *cube = scanctx->data;
 	Hyperspace *space = scanctx->space;
+	ChunkResult res = CHUNK_IGNORED;
 	int			i;
 
 	if (chunk->cube->num_slices != space->num_dimensions ||
 		!hypercubes_collide(cube, chunk->cube))
-		return true;
+		return CHUNK_IGNORED;
```

Comment:
With this change, the current chunk will not count towards `num_processed` in `foreach_chunk`. Is that intentional or should this be `CHUNK_PROCESSED`? (If leaving it as `CHUNK_IGNORED`, consider changing to `return res`.)

Reply:
Yes, this is intentional. I think it was wrong to return `true` here previously, although it didn't cause a problem since we didn't look at `num_processed` in this case. Now `num_processed` will correctly reflect the number of chunks that collided rather than "all" chunks.